scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,241 @@
+                DUMMY_CONF_FILES, '.travis_has_extra_test.yml'), 'r').read()
+        travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        return travis_ci_dict
+
+    def _return_empty_list(self):
+        return []
+
+    def test_read_protractor_file(self):
+        expected_protractor_config_file = python_utils.open_file(
+            os.path.join(
+                os.getcwd(), 'core', 'tests', 'protractor.conf.js'), 'r').read()
+        actual_protractor_config_file = (
+            check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(expected_protractor_config_file,
+                         actual_protractor_config_file)
+
+    def test_read_travis_ci_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(os.getcwd(), '.travis.yml'), 'r').read()
+        expected_travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        actual_travis_ci_dict = (
+            check_e2e_tests_are_captured_in_ci.read_and_parse_travis_yml_file())
+        self.assertEqual(expected_travis_ci_dict, actual_travis_ci_dict)
+
+    def test_dummy_read_protractor_file(self):
+        expected_protractor_config_file = (python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'dummy_protractor.conf.js'), 'r').read())
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file)
+        with dummy_path:
+            actual_protractor_config_file = (
+                check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(expected_protractor_config_file,
+                         actual_protractor_config_file)
+
+    def test_dummy_read_travis_ci_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+        expected_travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_ci_dict = (
+                check_e2e_tests_are_captured_in_ci
+                .read_and_parse_travis_yml_file())
+        self.assertEqual(expected_travis_ci_dict, actual_travis_ci_dict)
+
+    def test_get_e2e_suite_names_from_jobs_travis_yml_file(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_jobs = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_travis_jobs)
+
+    def test_get_e2e_suite_names_from_jobs_travis_yml_file_fail(self):
```

Comment:
This name needs more info. Fail in what way? Try to convey that more clearly using the variable names, method names, names of the files, etc. -- don't just say it fails. Is it supposed to throw an error or return the wrong result? Hard to tell.

At the moment, the test reads as though you're just reading a different file and getting different jobs extracted, which doesn't seem like a fail. Would expect to see some kind of exception with self.assertRaisesRegexp.

Ditto below.

Reply:
Yes I agree this test does not make much sense, removing that. 
Not needed. Done